.github/workflows/ci-schedule.yml

```diff
@@ -3,6 +3,10 @@ on:
   schedule:
     # Run this workflow "At 18:00 UTC on Sunday and Saturday"
     - cron: '0 18 * * 0,6'
+
+permissions:
+  actions: write # Required for dynamic workflow control, enabling retrying of job steps in case of failure (actions: nick-fields/retry@v3.0.0).
```

Comment:
@aditya7302 Whether write permission is required for action `nick-fields/retry` ? Refer to https://github.com/zhzhuang-zju/karmada/actions/runs/9758042445/job/26931766980, when I verify locally, I can enable retrying of job steps without write permission.

Reply:
@zhzhuang-zju after using the [StepSecurity's online tool](https://app.stepsecurity.io/secureworkflow/) for this workflow.
The following was the output 
```
name: FOSSA
on:
  push:
    # Exclude branches created by Dependabot to avoid triggering current workflow
    # for PRs initiated by Dependabot.
    branches-ignore:
      - 'dependabot/**'
permissions:  # added using https://github.com/step-security/secure-repo
  contents: read

jobs:
  fossa:
    name: FOSSA
    # prevent job running from forked repository, otherwise
    # 1. running on the forked repository would fail as missing necessary secret.
    # 2. running on the forked repository would use unnecessary GitHub Action time.
    if: ${{ github.repository == 'karmada-io/karmada' }}
    runs-on: ubuntu-22.04
    steps:
      - name: Harden Runner
        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
        with:
          egress-policy: audit

      - name: checkout code
        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
      - name: Run FOSSA scan and upload build data
        uses: fossas/fossa-action@f61a4c0c263690f2ddb54b9822a719c25a7b608f # v1.3.1
        with:
          api-key: ${{secrets.FOSSA_API_KEY}}
```
I think actions: write permissions is not required.